### PR TITLE
Wrap blockmean

### PR DESCRIPTION
**Pull request by meghanrjones**
**Description of proposed changes**

This PR wraps the [blockmean module](https://docs.generic-mapping-tools.org/dev/blockmean.html). Some changes involved in this PR:

- Most of the functionality from the `blockmedian` method is moved to `_blockm`
- `blockmedian` and `blockmean` call `_blockm` with the `block_method` parameter
- `pygmt/src/blockmedian.py` is renamed to `pygmt/src/blockm.py` for simplicity given the similarities between these three methods. I think this is simpler (more pythonic) than having each in separate modules and importing `_blockm` from a fourth module, but feedback is welcome if others feel that #807 should be more strictly adhered to.
- New tests for blockmean are added

Addresses #1091 

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/src/blockm.py
Comment (seisman) [inline pygmt/src/blockm.py:148]: Add a "Parameters" section and explain the "block_method" parameter.

Comment (meghanrjones) [reply pygmt/src/blockm.py:148]: Added in https://github.com/GenericMappingTools/pygmt/pull/1092/commits/62a763bc68a91e2521e9c77db74cd55696335f7a

File: pygmt/src/blockm.py
Comment (seisman) [inline pygmt/src/blockm.py:176]: Sorry, I didn't mean to explain all parameters. As this is an internal function, explaining the unique parameter `block_method` is enough. 
```suggestion
```

Comment (meghanrjones) [reply pygmt/src/blockm.py]: I read the comment in email and did not notice the in-line suggestion. The removal was done in https://github.com/GenericMappingTools/pygmt/pull/1092/commits/55a97be3aad14bd177fa56eb972ea2df244cc2a1

File: pygmt/tests/test_blockmean.py
Comment (weiji14) [inline pygmt/tests/test_blockmean.py:78]: One thing I've been contemplating is whether to just give people a `pandas.DataFrame` when no argument is passed into the `outfile` parameter. I feel like the current way is too restrictive, why can't people just pass in a filename (str) input to `table` and get a `pandas.DataFrame` directly? If they set an `outfile`, then return `None` by all means, but I don't think we should force `table=filename` inputs to always go into the `outfile`. What do you think?

Comment (seisman) [reply pygmt/tests/test_blockmean.py:78]: I agree it's too restrictive. The input could be a file, a matrix, a pandas.DataFrame or x/y/z.
The output could be a file or a pandas.DataFrame.

Comment (meghanrjones) [reply pygmt/tests/test_blockmean.py:78]: I also agree. Do you want me to add that to this PR? I would tend towards implementing it separately, since it will impact both blockmean and blockmedian.

Comment (seisman) [reply pygmt/tests/test_blockmean.py:78]: I was trying to improve the blockmean wrapper yesterday and realized that it will cause conflicts with this PR. I think we should improve it in a separate PR, either before or after this PR is merged.

Comment (meghanrjones) [reply pygmt/tests/test_blockmean.py:78]: After seems better to me since this PR is done as far as I can tell and that PR is not open yet. Do you want to work on improving the wrapper? I'm glad to help with adding DataFrame output support, but also fine with leaving it to you in which case I would wait until after you are done to open the PR for wrapping blockmode.

Comment (seisman) [reply pygmt/tests/test_blockmean.py:78]: I'll leave it to you or @weiji14, as I have never used blockm*.

Comment (meghanrjones) [reply pygmt/tests/test_blockmean.py:78]: Sounds good. I'll open an issue to keep track of this proposal.

File: pygmt/src/blockm.py
Comment (weiji14) [inline pygmt/src/blockm.py:140]: More just a matter of style, but I feel like this `_blockm` function should be placed near the top of the file (before the actual `blockmean` and `blockmedian` functions).

**Review by weiji14 — COMMENTED** [review]
Just one very minor point on moving `_blockm` to the top of the file, otherwise well done!

File: pygmt/src/blockm.py
Comment (meghanrjones) [reply pygmt/src/blockm.py]: OK, done in https://github.com/GenericMappingTools/pygmt/pull/1092/commits/f34e77842ab870e3124cf8b15df8a1080ab946f5

**Review by weiji14 — APPROVED** [review]
Thanks Meghan! Really appreciate having another data processing module added to PyGMT's toolkit :smile:

**Review by seisman — APPROVED** [review]
Looks good to me.

**Pull request merged**
